### PR TITLE
[RLlib] Fixed the autorom dependency issue

### LINKED ISSUE #31880: [Ray release infra] RLLib dependency on Atari ROMs causing multiple release test failures
Some release tests are failing because we're not able to build the cluster environment. The cluster environment logs fail when attempting to download an Atari ROM for RL simulations. 

Failing tests caused by this:
* long_running_many_tasks_serialized_ids
  * https://buildkite.com/ray-project/release-tests-branch/builds/1305#0185d81d-4369-4dbb-8a9b-bad7eae74509
  * cluster URL with failure logs https://console.anyscale.com/o/anyscale-internal/configurations/app-config-details/bld_4bepbq1l2i6parlr11maanru51
* long_running_many_actor_tasks
  * https://buildkite.com/ray-project/release-tests-branch/builds/1305#0185d81d-435d-45c2-a134-eca4da8e42a1
  * cluster URL with failure logs https://console.anyscale.com/o/anyscale-internal/configurations/app-config-details/bld_zskzkuz7w2p9j43lbch11n9qif
* long_running_actor_deaths
  * https://buildkite.com/ray-project/release-tests-branch/builds/1305#0185d81d-4355-498f-8ea6-1102a5ce29df
  * cluster URL with failure logs  https://console.anyscale.com/o/anyscale-internal/configurations/app-config-details/bld_4qcss9m5irrl35e5hbdfxvpazf
* rllib_learning_tests_ppo_tf
  * https://buildkite.com/ray-project/release-tests-branch/builds/1303#0185d157-4d6c-48e4-895c-1107461633b8
  * https://console.anyscale-staging.com/o/anyscale-internal/configurations/app-config-details/bld_bhxwsnz65j1k3myr5wwxebg2ae
* rllib_multi_gpu_learning_tests
  * https://buildkite.com/ray-project/release-tests-branch/builds/1303#0185d157-4d85-43b3-ae48-1993e127c46c
  * https://console.anyscale-staging.com/o/anyscale-internal/configurations/app-config-details/bld_hbd71md41tbzfi9crtv1mmhe5h

there may be others.

```
[ERROR] 1/21/2023, 10:28:56 PM:   error: subprocess-exited-with-error
  
  × Building wheel for AutoROM.accept-rom-license (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [241 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib
      copying AutoROM.py -> build/lib
      installing to build/bdist.linux-x86_64/wheel
      running install
      running install_lib
      creating build/bdist.linux-x86_64
      creating build/bdist.linux-x86_64/wheel
      copying build/lib/AutoROM.py -> build/bdist.linux-x86_64/wheel
      running install_egg_info
      running egg_info
      writing AutoROM.accept_rom_license.egg-info/PKG-INFO
      writing dependency_links to AutoROM.accept_rom_license.egg-info/dependency_links.txt
      writing requirements to AutoROM.accept_rom_license.egg-info/requires.txt
      writing top-level names to AutoROM.accept_rom_license.egg-info/top_level.txt
      reading manifest file 'AutoROM.accept_rom_license.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE.txt'
      writing manifest file 'AutoROM.accept_rom_license.egg-info/SOURCES.txt'
      Copying AutoROM.accept_rom_license.egg-info to build/bdist.linux-x86_64/wheel/AutoROM.accept_rom_license-0.5.4-py3.7.egg-info
      running install_scripts
      /tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/config/setupcfg.py:515: SetuptoolsDeprecationWarning: The license_file parameter is deprecated, use license_files instead.
        warnings.warn(msg, warning_class)
      time=1.0/60 seconds - Trying to download atari roms
          current status=downloading metadata (downloading_metadata)
          total downloaded bytes=0
          total payload download=0
          total failed bytes=0
<snip />
      time=11.0/60 seconds - Trying to download atari roms
          current status=download (downloading)
          total downloaded bytes=383148
          total payload download=381584
          total failed bytes=0
      Terminating attempt to download ROMs after 60 seconds, trying again
      AutoROM will download the Atari 2600 ROMs.
      They will be installed to:
          /tmp/pip-install-xrwywji6/autorom-accept-rom-license/build/bdist.linux-x86_64/wheel/AutoROM/roms
      
      Existing ROMs will be overwritten.
      Traceback (most recent call last):
        File "/home/ray/anaconda3/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 351, in <module>
          main()
        File "/home/ray/anaconda3/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 333, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/home/ray/anaconda3/lib/python3.7/site-packages/pip/_vendor/pep517/in_process/_in_process.py", line 250, in build_wheel
          metadata_directory)
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 414, in build_wheel
          wheel_directory, config_settings)
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 398, in _build_with_temp_dir
          self.run_setup()
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 485, in run_setup
          self).run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/build_meta.py", line 335, in run_setup
          exec(code, locals())
        File "<string>", line 20, in <module>
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/__init__.py", line 108, in setup
          return distutils.core.setup(**attrs)
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/_distutils/core.py", line 185, in setup
          return run_commands(dist)
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
          dist.run_commands()
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
          self.run_command(cmd)
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/dist.py", line 1213, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/wheel/bdist_wheel.py", line 360, in run
          self.run_command("install")
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/_distutils/cmd.py", line 318, in run_command
          self.distribution.run_command(command)
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/dist.py", line 1213, in run_command
          super().run_command(command)
        File "/tmp/pip-build-env-pco1a5lu/overlay/lib/python3.7/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
          cmd_obj.run()
        File "<string>", line 15, in run
        File "/tmp/pip-install-xrwywji6/autorom-accept-rom-license/AutoROM.py", line 330, in main
          buffer = torrent_tar_to_buffer()
        File "/tmp/pip-install-xrwywji6/autorom-accept-rom-license/AutoROM.py", line 200, in torrent_tar_to_buffer
          raise RuntimeError("Tried to download ROMs 3 times, which have all failed, please try again or report this issue.")
      RuntimeError: Tried to download ROMs 3 times, which have all failed, please try again or report this issue.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.

[ERROR] 1/21/2023, 10:28:56 PM:   ERROR: Failed building wheel for AutoROM.accept-rom-license
```

### PR DESCRIPTION
**Pull request by kouroshHakha**
Signed-off-by: Kourosh Hakhamaneshi <kourosh@anyscale.com>

## Why are these changes needed?
Closes https://github.com/ray-project/ray/issues/31880

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kouroshHakha** [comment]
build kite: https://buildkite.com/ray-project/release-tests-pr/builds/26418

File: release/rllib_tests/app_config.yaml [hunk 1]
```diff
@@ -19,7 +19,9 @@ python:
   - ale-py==0.8.0
   - gym==0.26.2
   - mujoco-py<2.2,>=2.1
-  - autorom[accept-rom-license]
+  # AutoROM wheels are sometimes not available on PyPI, so we use a pre-built wheel 
+  # stored in our S3 bucket. This wheel was tested for python 3.7, 3.8, and 3.9.
```
Comment (cadedaniel) [inline release/rllib_tests/app_config.yaml:23]:
```suggestion
  # AutoROM downloads ROMs via torrent when they are built. The torrent is unreliable,
  # so we built it for py3 and use that instead. This wheel was tested for python 3.7, 3.8,
  # and 3.9.
```

**Review by cadedaniel — APPROVED** [review]
suggestion on wording

File: release/rllib_tests/app_config.yaml [hunk 2]
```diff
@@ -19,7 +19,10 @@ python:
   - ale-py==0.8.0
   - gym==0.26.2
   - mujoco-py<2.2,>=2.1
-  - autorom[accept-rom-license]
+  # AutoROM downloads ROMs via torrent when they are built. The torrent is unreliable,
+  # so we built it for py3 and use that instead. This wheel was tested for python 3.7, 3.8,
+  # and 3.9.
+  - https://ray-ci-deps-wheels.s3.us-west-2.amazonaws.com/AutoROM.accept_rom_license-0.5.4-py3-none-any.whl
```
Comment (cadedaniel) [inline release/rllib_tests/app_config.yaml:25]: one thing, do we need to also install `autorom`? can't remember if this installs `autorom` or just the actual ROMs..

Comment (kouroshHakha) [reply release/rllib_tests/app_config.yaml:25]: This should be sufficient. Let's see if the release tests pass now on ci.

**Comment by cadedaniel** [comment]
Can we merge? https://buildkite.com/ray-project/release-tests-pr/builds/26419#0185eab9-a340-4114-a845-56d8aa79173e passed

**Comment by kouroshHakha** [comment]
@cadedaniel Do we know why the other test have not started yet? Can we increase the available compute for the release?

**Comment by kouroshHakha** [comment]
> Can we merge? https://buildkite.com/ray-project/release-tests-pr/builds/26419#0185eab9-a340-4114-a845-56d8aa79173e passed

We need more data point :)

**Comment by cadedaniel** [comment]
Oh, I thought you only specified one run. Hmm capacity yeah.. did you run with high priority?

I think this should be high priority given that it's hard to get signal from many release tests until this is fixed

**Comment by kouroshHakha** [comment]
> Oh, I thought you only specified one run. Hmm capacity yeah.. did you run with high priority?
> 
> I think this should be high priority given that it's hard to get signal from many release tests until this is fixed

Oh I didn't set the priority. running it again. :(

**Comment by kouroshHakha** [comment]
How long should it take with high priority? it's still not staring the tests?

**Comment by cadedaniel** [comment]
Not sure, perhaps it is simply next in line? We might miss the nightly run tonight, it's OK.

**Comment by cadedaniel** [comment]
Cc @krfricke do you know why the tests are still pending? Maybe we should just merge..

**Comment by kouroshHakha** [comment]
> Cc @krfricke do you know why the tests are still pending? Maybe we should just merge..

@krfricke just to provide more context most of the test that are pending require only 1gpu_16cpu (g3.4xlarge) machines

**Comment by kouroshHakha** [comment]
@cadedaniel I think we should merge. The failures are the known ones. @ArturNiederfahrenhorst has a PR https://github.com/ray-project/ray/pull/31910 would most likely bring down the flakiness of Impala tests.

**Comment by cadedaniel** [comment]
Sounds good! LGTM

**Pull request merged**